Society Rework

zumbak04 opened the pull request:
**NOTE FOR TESTERS: Test all society interactions: quests, recruiting and abduction (if it's the Cult of the Damned, the Shadow Council, the Twilight's Hammer), joining, ranking up and features from "Public changelog". I think it's enough.**

Public changelog:
- Kaldorei females no longer can be druids.
- Kaldorei priests, druids and Ancients of Lore can be Court Chaplains of Kaldorei liege.
- Kaldorei characters can ask Kaldorei priests to be their mentors.
- The Cenarion Circle and the Sisterhood of Elune no longer require to be druid/priest to join (not rank up), but require 10 Learning or Learning education.
- Members of the Cenarion Circle and the Sisterhood of Elune don't need Learning to become druids/priests.
- Enter the Emerald Dream and Build a Hospital quests give more society currency.
- Kaldorei can Order to Take Vows to send character either to the Sisterhood of Elune or the Cenarius Circle and remove from line of inheritance.
- Lowered opinion of other members in the Cenarion Circle and the Sisterhood of Elune.
- AI members of the Cenarion Circle and the Sisterhood of Elune won't do hostile actions towards other members.

Short secret changelog:
- Reworked all society interactions: ranking up, recruiting and abduction (if it's the Cult of the Damned, the Shadow Council, the Twilight's Hammer), quests, welcome messages, conditions to join and show society to improve readability and delete as much duplicated lines as possible.
- Society-related triggers, effects, quests has been moved to new files: `events/wc_quest_events.txt`, `common/scripted_triggers/wc_society_scripted_triggers.txt`, `common/scripted_effects/wc_society_scripted_effects.txt`.

Full secret changelog (reworking):
- Refactored rank up in the Cenarion Circle and the Sisterhood of Elune.
- Purged separate rank up events and reworked them into one.
- Reworked all society conditions (to join society, to show society etc.).
- Merged all recruit decisions into one located in `mnm_devil_worshipers_events.txt`.
- Reworked society recruitment localization.
- Created `ai_join_random_society_effect`, `join_interesting_society_effect` effects and `is_interested_in_joining_trigger` trigger out of code of `MNM.10011` and `MNM.10031` events to simplify adding of new societies.
- Merged `teach_yourself_priest` and `become_priestess_of_elune` decisions.
- Removed `WCCC.100` event that forced AI to join the CC because it was laggy.
- Merged society welcome messages into vanilla chain in mnm_monastic_orders_events.txt.
- Added customized dynamic localization in both English and German localization of SoE and CC quests and ranking up event.
- AI characters get druid/priest class automatically joining CC and SoE societies.
- Lea Stonepaw (female night elf) isn't druid in 583 and 603 bookmarks.
- Replaced `society_member_of = monastic_order_stoics` by `is_member_of_stoics_trigger` for future use.
- Partially adapted `mnm_monastic_orders_events.txt` for future use.
- Merged `send_to_temple` into `order_to_take_vows`.
- Made all societies indestructible.

Full secret changelog (moving):
- Moved society triggers to `common/scripted_triggers/wc_society_scripted_triggers.txt`.
- Moved society effects to `common/scripted_effects/wc_society_scripted_effects.txt`.
- Moved society quests to `events/wc_quest_events.txt`.

MemeWizard commented on localisation/00_wc_sad.csv:
```diff
@@ -110,7 +110,7 @@ build_cenarion_enclave_desc;Invite your fellow members to create a Cenarion Encl
 raise_host_of_the_wilds_desc;The Cenarion Circle has many powerful creatures of the forests among its members. Now in times of war you can unleash them upon your enemies.;;Der Zirkel des Cenarius hat viele m�chtige Waldgesch�pfe unter seinen Mitgliedern. Jetzt, in der Zeit des Krieges, k�nnt Ihr sie gegen die Feinde Azeroths einsetzen.;;;;;;;;;;;x
 ;;;;;;;;;;;;;;x
-cenarion_circle;Cenarion Circle;;Zirkel des Cenarius;;;;;;;;;;;x
+cenarion_circle;the Cenarion Circle;;Zirkel des Cenarius;;;;;;;;;;;x
```
the the

<details>

![image](https://user-images.githubusercontent.com/46576860/64913551-bd00aa80-d74a-11e9-9923-3d763853a04f.png)

</details>

zumbak04 replied:
Fixed!

MemeWizard commented on localisation/00_wc_sad.csv:
```diff
@@ -119,12 +119,12 @@ cenarion_circle_rank_2_male;Hierophant;;Hierophant;;;;;;;;;;;x
+cenarion_circle_rank_5_female;Archdruid;;Erzdruidin;;;;;;;;;;;x
 ;;;;;;;;;;;;;;x
-sisterhood_of_elune;Sisterhood of Elune;;Schwesternschaft von Elune;;;;;;;;;;;x
+sisterhood_of_elune;the Sisterhood of Elune;;Schwesternschaft von Elune;;;;;;;;;;;x
```
the the

<details>

![image](https://user-images.githubusercontent.com/46576860/64913663-f20dfc80-d74c-11e9-8419-0597f6b8074d.png)

</details>

zumbak04 replied:
Fixed!

MemeWizard commented on localisation/00_zuwc_united_localisation.csv:
```diff
@@ -2981,12 +2965,19 @@ GetHorses_horses;horses;;;;;;;;;;;;;x
+EVTDESC_MNM_7022_B;We know what has happened, [Root.GetSocietyRank] [Root.GetFirstName]\nThe recruit you sent us, [satanists_recruit.GetTitledFirstName], has proven to be disaster. We have spent significant resources in our attempt to induct [satanists_recruit.GetHerHim], all for naught. This lack of judgment will be remembered.\n\nSigned, [From.GetSocietyRank] [From.GetFirstName];;;;;;;;;;;;;x
+EVTDESC_MNM_4452;[From.GetSocietyGreeting], [Root.GetTitledFirstName]\n\nWe are delighted by your decision to join the ranks of [From.Society.GetName], [Root.GetTitle]. [From.GetSocietyGoal].\n\n[From.GetSocietyFarewell],\nSigned, [From.GetSocietyRank] [From.GetFirstName];;;;;;;;;;;;;x
+GetSocietyGreeting_cenarion_circle;Ishnu'alah;;;;;;;;;;;;;x
+GetSocietyGreeting_sisterhood_of_elune;Ishnu'alah;;;;;;;;;;;;;x
```
> Ishnu'alah

What if CC/SoE leader culture is not Kaldorei?

<details>

![image](https://user-images.githubusercontent.com/46576860/64915565-af5f1b00-d772-11e9-8c12-fe038a5b56e4.png)

</details>

zumbak04 replied:
Shame on those who do not know what this phrase means. It's absolute legend!

zumbak04 replied:
Replaced...
**SAD**

MemeWizard replied:
> Replaced...
**SAD**

You know, you can just add another Society Greeting if society leader is not kaldorei instead of replacing base greeting with generic

zumbak04 replied:
No, Ishnu'alah is already dead.

MemeWizard replied:
**SAD**

MemeWizard commented:
>The Cenarion Circle and the Sisterhood of Elune no longer require to be druid/priest to join (not rank up), but require 10 Learning or Learning education.

Now there are furbolgs/humans in CC, who are not even druids

> Kaldorei females no longer can be druids.

> Many years ago, the two genders were even forbidden to take the opposite role. When this was finally lifted, it was a tremendous and tumultuous societal change, with a pair of statues created to commemorate the first male priest and the first female druid.

How about ban non-night elves in CC/SoE and NE female druids/male priests until NE lose their immortality after fall of Nordrassil?

zumbak04 commented:
> How about ban non-night elves in CC/SoE and NE female druids/male priests until NE lose their immortality after fall of Nordrassil?

I would say it happened because they joined the Alliance (at least xenophobic part).
It requires design and big narrative event so make suggestion.

zumbak04 commented on localisation/00_zuwc_united_localisation.csv:
```diff
@@ -536,40 +536,24 @@ EVTDESC_WCPOD_5013_plague;Your days in a living body have come to the end. Your
-EVTOPTB_WCCSO_7004;It was interesting, but I do not think [From.GetSheHe] is ready for more.;;Es war interessant, aber ich glaube nicht, dass [From.GetSheHe] zu mehr bereit ist.;;;;;;;;;;;x
-EVTOPTC_WCCSO_7004;Time is wasted!;;Was f�r eine Zeitverschwendung!;;;;;;;;;;;x
-EVTDESC_WCCSO_7005;Soon, the strange [From.GetWomanMan] comes to meet you. Coming to you, [From.GetSheHe] just hands you a letter and disappears around the nearest corner. The letter is unfamiliar and in a strange print. After reading it, you realize that it is an invitation to a society by the name of [From.Society.GetName].;;Schon bald kam [From.Geteineine] [From.GetWomanMan] Euch zu besuchen.  ...
+EVTDESC_WCCSO_7001_case_01;I'm having dinner and plenty of drinks with [satanists_recruiter.GetTitledName] tonight. It's lovely, yet what I am most charmed by is [satanists_recruiter.GetFirstName] [satanists_recruiter.GetHerselfHimself]. Why else would I not immediately stop our conversation, which has slowly begun to touch upon some unsavory and deviant topics about our society?;;;;;;;;;;;;;x
```
Used customized localization here.

zumbak04 commented:
> Now there are furbolgs/humans in CC, who are not even druids

Shoud be fixed.

MemeWizard commented:
get rid of who

<details>

![image](https://user-images.githubusercontent.com/46576860/64943312-144e6a00-d874-11e9-8f32-ba30602495fc.png)

</details>

MemeWizard commented:
> Kaldorei females no longer can be druids.

But male kaldorei still can be priest

<details>

![image](https://user-images.githubusercontent.com/46576860/64953880-b5e1b580-d88c-11e9-9b09-e483adbb39a0.png)

</details>

MemeWizard commented:
> Kaldorei females no longer can be druids.

But furbolg females is ok?

<details>

![image](https://user-images.githubusercontent.com/46576860/64954630-d90d6480-d88e-11e9-9735-949843d04532.png)

</details>

zumbak04 commented:
> > Kaldorei females no longer can be druids.
> 
> But male kaldorei still can be priest

Only females can hold temples now.

> > Kaldorei females no longer can be druids.
> 
> But furbolg females is ok?

If she became druid not being Kaldorei, yep.

geffe1 commented:
>Only females can hold temples in Kaldorei religion.

Ok.. I see the temple being in a man's hands here.. right.. there.
![ck2_359](https://user-images.githubusercontent.com/31706375/64982661-03771600-d8bf-11e9-9efd-0d160b986789.png)

geffe1 commented:
![ck2_362](https://user-images.githubusercontent.com/31706375/64983135-1ccc9200-d8c0-11e9-9081-f84c04e87c0b.png)
This summons a druid- should probably be reworked to summon a female kaldorei priestess instead.

zumbak04 commented:
> Ok.. I see the temple being in a man's hands here.. right.. there. Can not hand them out to his male children though.. so it works.. somewhat, though we probably need to clean that up so that only women holds the temples.

Fixed in https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/pull/722

> This summons a druid- should probably be reworked to summon a female kaldorei priestess instead, as men can not be the court chaplains. And be renamed to "Invite Holy Woman to court"

Implemented... also, male druids can be Court Chaplains now.

MemeWizard commented:
Is cultist suspicion mechanic doesn't work?
I tried a lot of secret activity, but still didn't get "suspicions to belong to a secret society" modifier

MemeWizard commented:
Looks like I can't complete this quest
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46576860/65814502-620b9080-e1eb-11e9-919f-8e448c04aa35.png)

</details>

[Mechagon621_05_04.zip](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/files/3665273/Mechagon621_05_04.zip)

zumbak04 commented:
> Looks like I can't complete this quest
> 
> Click to expand
> [Mechagon621_05_04.zip](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/files/3665273/Mechagon621_05_04.zip)

Can't reproduce it. Looks like vanilla bug because no one touched insides of this event chain in the mod.

> Is cultist suspicion mechanic doesn't work?
> I tried a lot of secret activity, but still didn't get "suspicions to belong to a secret society" modifier

Report is as separate issue.

MemeWizard commented:
Looks like there is no Cult of the Twillight's Hammer in 603 bookmark

<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46576860/65939855-b0a96c80-e42f-11e9-822d-e95508db03a0.png)

</details>

zumbak04 commented:
> Looks like there is no Cult of the Twillight's Hammer in 603 bookmark

Fixed!

MemeWizard commented:
> Only females can hold temples in Kaldorei religion.

<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46576860/65997292-c3f52000-e4a1-11e9-9e1a-b1997bd53b53.png)

</details>

zumbak04 commented:
> > Only females can hold temples in Kaldorei religion.

> > Kaldorei females no longer can be druids.
> 
> But male kaldorei still can be priest

That's just localization not related to classes (Priest and Druid) or societies so I moved it to separate branch.

geffe1 commented:
![ck2_386](https://user-images.githubusercontent.com/31706375/66703183-13dcae00-ed10-11e9-921f-9735109ca401.png)
Papacy?

![ck2_387](https://user-images.githubusercontent.com/31706375/66703191-20610680-ed10-11e9-8306-1de8d1877ebc.png)
Monastries? I thought we only had temples!

![ck2_388](https://user-images.githubusercontent.com/31706375/66703195-38388a80-ed10-11e9-9c69-5ccef3d1beb3.png)
I think "Priestess" would be a better option if we keep female restricted to being the only priests in the Kaldorei society. Otherwise it works fine, if we plan to make events for males to be able to become it later on.

zumbak04 commented:
> Papacy?

Report? in separate issue, it's not related to the society rework.

> Monastries? I thought we only had temples!

Suggest dynamic localization in separate issue, it's not related to the society rework.

> I think "Priestess" would be a better option if we keep female restricted to being the only priests in the Kaldorei society. Otherwise it works fine, if we plan to make events for males to be able to become it later on.

REE! They're able to become Court Chaplains.
